scripts/simplify_network.py

```diff
@@ -428,6 +429,9 @@ def split_links(nodes):
                 p_min_pu=-p_max_pu,
                 underground=False,
                 under_construction=False,
+                geometry_links=LineString(
+                    zip(n.buses.loc[[b[0], b[1]]].x, n.buses.loc[[b[0], b[1]]].y)
+                ),
```

Comment:
have you tried to name "geometry" instead of "geometry_links"?
The error seemed related to the improper defintion of the column geometry, overriding it should do the trick, but not sure...

for styling [b[0], b[1]] may be replaced with b[0:2] or b itself if it contains only 2 elements

Reply:
Yes, absolutely! An initial version has used `geometry` as a column name, but it looks like it's being transformed into `'links_geometry'` by some PyPSA method when exporting to nc. Which leads to the mentioned `xarray cannot serialize arbitrary Python objects`. Do I miss something?

Done for indexing re-styling :)